Scripts/build-archive.sh

```diff
@@ -52,6 +55,33 @@ if [ ${#invalid_bitcode[@]} -ne 0 ]; then
   exit 1
 fi
 
+# Verify architectures.
+function verify_framework_architectures() {
+  local name=${1##*/}
+  name=${name%.*}
+  local archs=($(lipo -archs "$1/$name"))
+  archs=($(printf '%s\n' "${archs[@]}" | sort))
+  required=($(printf '%s\n' "${@:2}" | sort))
+  if [[ "${archs[@]}" != "${required[@]}" ]]; then
+    echo "${1#$PRODUCTS_DIR/} doesn't contain required architectures. It has '${archs[@]}' but '${required[@]}' are required."
+    return 1
+  fi
+}
+function verify_architectures() {
```

Comment:
Is it going to be used anywhere else? Currently the only use is in frameworks, and this specific method looks redundant

Reply:
Only one use, but it's logically separated, so it's OK